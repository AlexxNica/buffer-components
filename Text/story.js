import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Text from './index';

const text = 'The quick brown fox jumps over the lazy dog';

storiesOf('Text')
  .add('Default', () => (
    <Text>{text}</Text>
  ))
  .add('Color', () => (
    <Text color={"#329ced"}>{text}</Text>
  ))
  .add('Large', () => (
    <Text size={36}>{text}</Text>
  ))
  .add('Small', () => (
    <Text size={10}>{text}</Text>
  ))
  .add('Bold', () => (
    <Text weight={800}>{text}</Text>
  ))
  .add('Thin', () => (
    <Text weight={200}>{text}</Text>
  ));
