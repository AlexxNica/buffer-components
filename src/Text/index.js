import React, { PropTypes } from 'react';
import reactCSS from 'reactcss';

const Text = ({ children, color, size, weight }) => {
  const styles = reactCSS({
    default: {
      Text: {
        fontSize: size,
        color,
        fontFamily: '"Open Sans", sans-serif',
        fontWeight: weight,
      },
    },
  });
  return (
    <span style={styles.Text}>{children}</span>
  );
};

Text.propTypes = {
  children: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.number,
  weight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Text.defaultProps = {
  size: 14,
  weight: 400,
};

export default Text;
