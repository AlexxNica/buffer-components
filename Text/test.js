jest.unmock('./index');
import React from 'react';
import { renderAndCheckSnapshot } from '../testHelpers';
import Text from './index';

describe('Text', () => {
  it('Text component', () => {
    renderAndCheckSnapshot(
      <Text>Howdy!</Text>
    );
  });

  it('Text w/ Color', () => {
    renderAndCheckSnapshot(
      <Text color={"#329ced"}>Howdy! (In Color)</Text>
    );
  });

  it('Text w/ Size', () => {
    renderAndCheckSnapshot(
      <Text size={36}>Big Howdy!</Text>
    );
  });

  it('Text w/ Weight', () => {
    renderAndCheckSnapshot(
      <Text weight={600}>Big Howdy!</Text>
    );
  });
});
