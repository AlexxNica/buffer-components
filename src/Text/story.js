import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Text from './index.js';

const text = 'The quick brown fox jumps over the lazy dog';

storiesOf('Text')
  .add('Default', () => (
    <Text>{text}</Text>
  ))
  .add('Text with color', () => (
    <Text color={"#329ced"}>{text}</Text>
  ));
