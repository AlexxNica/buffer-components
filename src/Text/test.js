jest.unmock('./index');
import React from 'react';
import Text from './index';
import renderer from 'react-test-renderer';

describe('Text', () => {
  it('Renders a plain Text component', () => {
    const component = renderer.create(
      <Text>Howdy!</Text>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders Text with some color', () => {
    const component = renderer.create(
      <Text color={"#329ced"}>Howdy! (In Color)</Text>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
