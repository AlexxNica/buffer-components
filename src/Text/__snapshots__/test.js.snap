exports[`Text Renders a plain Text component 1`] = `
<span>
  Howdy!
</span>
`;
